### PR TITLE
Bump minikube iso to v1.0.1, the buildroot based ISO

### PR DESCRIPTION
**Pull request by r2d4**
This makes the buildroot image the default image.  It removes the buildroot image integration test as all tests will run with this iso now.

The iso is now versioned so that everything <v1.0.0 is the b2d image and everything new is the buildroot image

Fixes https://github.com/kubernetes/minikube/issues/571

### CODE REVIEW

**Comment by dlorenc** [comment]
@minikube-bot test this please

**Comment by r2d4** [comment]
The OSX flakes are because I forgot to upgrade the xhyve driver.  Need to investigate the Linux ones

**Comment by dlorenc** [comment]
Does this need to be rebased for the Windows tests?

**Comment by r2d4** [comment]
No, it contains the latest windows fixes

File: README.md [hunk 1]
```diff
@@ -99,7 +99,6 @@ To use [rkt](https://github.com/coreos/rkt) as the container runtime run:
 $ minikube start \
     --network-plugin=cni \
     --container-runtime=rkt \
-    --iso-url=http://storage.googleapis.com/minikube/iso/buildroot/minikube-v0.0.6.iso
 ```
 
 This will use an alternative minikube ISO image containing both rkt, and Docker, and enable CNI networking.
```
Comment (syhol) [inline README.md]: It won’t use an "alternative minikube ISO image" anymore as the minikube-iso will be the default image.

Comment (r2d4) [inline README.md]: Good catch! I removed that line

File: hack/jenkins/osx_integration_tests_virtualbox_systemd.sh [hunk 2]
```diff
@@ -1,56 +0,0 @@
-#!/bin/bash
```
Comment (dlorenc) [inline hack/jenkins/osx_integration_tests_virtualbox_systemd.sh]: Could you also drop the pending job here: https://github.com/kubernetes/minikube/blob/master/hack/jenkins/minikube_cross_build_and_upload.sh

Comment (r2d4) [inline hack/jenkins/osx_integration_tests_virtualbox_systemd.sh]: Done

**Comment by dlorenc** [comment]
Would you mind rebasing? Anything else we need to take care of here?

**Comment by r2d4** [comment]
Rebased.  The only thing left to figure out is the flakes we've been seeing while switching between ISOs.  We should make sure we at least identify the cause of that before we switch over.

**Comment by dlorenc** [comment]
@minikube-bot test this please

**Comment by jonboulle** [comment]
@dlorenc what do we need to do to pick this up again?

**Comment by dlorenc** [comment]
@jonboulle we had one more flake to chase down, but I think with #943 we should be good to merge this.

**Comment by dlorenc** [comment]
@minikube-bot test this please

**Pull request merged**